comps/dataprep/minio/milvus/langchain/README.md

```diff
@@ -0,0 +1,252 @@
+
+```bash
+cd ../../..
+docker build --build-arg http_proxy=$http_proxy --build-arg https_proxy=$https_proxy -t opea/embedding-mosec-endpoint:latest -f comps/embeddings/mosec/langchain/dependency/Dockerfile .
+```
+
+Then start the mosec embedding server.
+
+```bash
+your_port=6010
+docker run -d --name="embedding-mosec-endpoint" -p $your_port:8000  opea/embedding-mosec-endpoint:latest
+```
+
+Setup environment variables:
+
+```bash
+export MOSEC_EMBEDDING_ENDPOINT="http://localhost:$your_port"
+export MILVUS_HOST=${your_host_ip}
+```
+
+### 1.5 Start Document Preparation Microservice for Milvus with Python Script
+
+Start document preparation microservice for Milvus with below command.
+
+```bash
+python prepare_doc_milvus.py
+```
+
+## 🚀2. Start Microservice with Docker (Option 2)
+
+### 2.1 Start Milvus Server
+
+Please refer to this [readme](../../../../vectorstores/milvus/README.md).
+
+### 2.2 Build Docker Image
+
+```bash
+cd ../../../..
+# build mosec embedding docker image
+docker build --build-arg http_proxy=$http_proxy --build-arg https_proxy=$https_proxy -t opea/embedding-langchain-mosec-endpoint:latest -f comps/embeddings/mosec/langchain/dependency/Dockerfile .
+# build dataprep milvus docker image
+docker build -t opea/dataprep-milvus:latest --build-arg https_proxy=$https_proxy --build-arg http_proxy=$http_proxy --build-arg no_proxy=$no_proxy -f comps/dataprep/milvus/langchain/Dockerfile .
+```
+
+### 2.3 Setup Environment Variables
+
+```bash
+export MOSEC_EMBEDDING_ENDPOINT="http://localhost:$your_port"
+export MILVUS_HOST=${your_host_ip}
+```
+
+### 2.3 Run Docker with CLI (Option A)
+
+```bash
+docker run -d --name="dataprep-milvus-server" -p 6010:6010 --ipc=host -e http_proxy=$http_proxy -e https_proxy=$https_proxy -e no_proxy=$no_proxy -e MOSEC_EMBEDDING_ENDPOINT=${MOSEC_EMBEDDING_ENDPOINT} -e MILVUS_HOST=${MILVUS_HOST} opea/dataprep-milvus:latest
+```
+
+### 2.4 Run with Docker Compose (Option B)
+
+```bash
+mkdir model
+cd model
+git clone https://huggingface.co/BAAI/bge-base-en-v1.5
+cd ../
+# Update `host_ip` and  `HUGGINGFACEHUB_API_TOKEN` in set_env.sh
+. set_env.sh
+docker compose -f docker-compose-dataprep-milvus.yaml up -d
+```
+
+## 🚀3. Consume Microservice
+
+### 3.1 Consume Upload API
+
+Once document preparation microservice for Milvus is started, user can use below command to invoke the microservice to convert the document to embedding and save to the database.
+
+Make sure the file path after `files=@` is correct.
+
+- Single file upload
+
+```bash
+curl -X POST \
+    -H "Content-Type: multipart/form-data" \
+    -F "files=@./file.pdf" \
+    http://localhost:6010/v1/dataprep
+```
+
+You can specify chunk_size and chunk_size by the following commands. To avoid big chunks, pass a small chun_size like 500 as below (default 1500).
+
+```bash
+curl -X POST \
+    -H "Content-Type: multipart/form-data" \
+    -F "files=@./file.pdf" \
+    -F "chunk_size=500" \
```

Comment:
Will it matter that a prompt might be chunked differently? What are the defaults for chunking and overlap between chunks?

Reply:
default chunk size seems to be `1500` and the overlap is `100`. I don't think the prompt will be chunked